packages/wrapper-react/src/index.tsx

```diff
@@ -47,31 +52,47 @@ export const MonacoEditorReactComp: React.FC<MonacoEditorProps> = (props) => {
 
     const apiWrapperRef = useRef<MonacoVscodeApiWrapper>(new MonacoVscodeApiWrapper(vscodeApiConfig));
     const haveEditorService = useRef(true);
+    const currentEditorConfig = useRef<EditorAppConfig | undefined>(undefined);
     const editorAppRef = useRef<EditorApp>(null);
-    const lcsManagerRef = useRef<LanguageClientsManager>(null);
     const containerRef = useRef<HTMLDivElement>(null);
     const onTextChangedRef = useRef(onTextChanged);
-    const [modifiedCode, setModifiedCode] = useState(modifiedTextValue);
-    const [originalCode, setOriginalCode] = useState(originalTextValue);
+    const modifiedCode = useRef<string>(modifiedTextValue);
+    const originalCode = useRef<string>(originalTextValue);
 
-    const runQueue = useRef<Array<() => Promise<void>>>([]);
+    const runQueue = useRef<Array<{id: string, func: () => Promise<void>}>>([]);
+    let queueAwait: Promise<void> | undefined = undefined;
+    let queueResolve: ((value: void | PromiseLike<void>) => void) | undefined = undefined;
 
-    const executeQueue = (id: string, newfunc: () => Promise<void>) => {
+    const addQueue = (id: string, func: () => Promise<void>) => {
         debugLogging(`Adding to queue: ${id}`);
         debugLogging(`QUEUE SIZE before: ${runQueue.current.length}`);
-        runQueue.current.push(newfunc);
-        (async () => {
-            // always expect to need to await the global init
-            await awaitGlobal();
-
-            while (runQueue.current.length > 0) {
-                const func = runQueue.current.shift();
-                debugLogging('QUEUE FUNC start', true);
-                await func?.();
-                debugLogging('QUEUE FUNC end');
+        runQueue.current.push({id, func});
+    };
+
+    const triggerQueue = () => {
+        setInterval(() =>  {
+            if (queueAwait === undefined) {
+                queueAwait = new Promise<void>((resolve) => {
+                    queueResolve = resolve;
+                });
+                executeQueue();
+            }
+        }, 50);
+    };
```

Comment:
I'm confused by that code :thinking: 

- it uses a local variable which is destroyed at each render
- nothing stops the interval :thinking: